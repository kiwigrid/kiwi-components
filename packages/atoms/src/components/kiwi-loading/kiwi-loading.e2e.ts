import { newE2EPage } from '@stencil/core/testing';

describe('kiwi-loading', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<kiwi-loading></kiwi-loading>');

    const element = await page.find('kiwi-loading');
    expect(element).toHaveClass('hydrated');
  });
});
