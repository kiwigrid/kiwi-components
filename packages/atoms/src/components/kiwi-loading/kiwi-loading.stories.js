import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Loading',
  parameters: { docs: { description: { component: notes } } },
  decorators: [],
};

export const basic = (args) => {
  const { loading, text } = args;

  return html` <div class="panel panel-default">
    <div class="panel-body">
      <kiwi-loading loading=${loading} text=${text} />
    </div>
  </div>`;
};
basic.args = {
  loading: true,
  text: 'Loading...',
};
