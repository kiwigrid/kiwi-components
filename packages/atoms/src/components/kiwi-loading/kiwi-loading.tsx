import {
  Component,
  ComponentInterface,
  h,
  Host,
  JSX,
  Prop,
} from '@stencil/core';

@Component({
  tag: 'kiwi-loading',
  shadow: false,
})
export class KiwiLoading implements ComponentInterface {
  /**
   * Indicates the loading state.
   */
  @Prop({ reflect: true })
  loading = true;

  /**
   * The label of the loading spinner.
   */
  @Prop()
  text!: string;

  render(): JSX.Element {
    return (
      this.loading && (
        <Host class="text-info">
          <div class="loading-spinner pull-left mr-xs-1">
            <div class="spinner small inline-block" />
          </div>
          <span>{this.text}</span>
        </Host>
      )
    );
  }
}
