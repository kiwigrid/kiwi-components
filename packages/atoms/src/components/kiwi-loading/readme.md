# kiwi-loading



<!-- Auto Generated Below -->


## Properties

| Property            | Attribute | Description                       | Type      | Default     |
| ------------------- | --------- | --------------------------------- | --------- | ----------- |
| `loading`           | `loading` | Indicates the loading state.      | `boolean` | `true`      |
| `text` _(required)_ | `text`    | The label of the loading spinner. | `string`  | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
