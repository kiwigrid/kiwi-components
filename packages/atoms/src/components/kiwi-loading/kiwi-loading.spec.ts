import { KiwiLoading } from './kiwi-loading';
import { newSpecPage } from '@stencil/core/testing';

describe('kiwi-loading', () => {
  it('renders', async () => {
    const loading = await newSpecPage({
      components: [KiwiLoading],
      html: `
        <kiwi-loading text="Windows is starting"></kiwi-loading>
      `,
    });

    expect(loading.root).toMatchSnapshot('loading');

    (loading.root as HTMLKiwiLoadingElement).loading = false;
    await loading.waitForChanges();

    expect(loading.root).toMatchSnapshot('not loading');
  });
});
