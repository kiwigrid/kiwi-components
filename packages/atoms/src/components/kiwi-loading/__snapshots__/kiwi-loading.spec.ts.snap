// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-loading renders: loading 1`] = `
<kiwi-loading class="text-info" loading="" text="Windows is starting">
  <div class="loading-spinner mr-xs-1 pull-left">
    <div class="inline-block small spinner"></div>
  </div>
  <span>
    Windows is starting
  </span>
</kiwi-loading>
`;

exports[`kiwi-loading renders: not loading 1`] = `<kiwi-loading text="Windows is starting"></kiwi-loading>`;
